kafka-rest/src/test/java/io/confluent/kafkarest/integration/AbstractConsumerTest.java

```diff
@@ -184,12 +209,42 @@ protected String startConsumeMessages(
     Response response = request(instanceUri + "/records").accept(accept).get();
     assertOKResponse(response, responseMediatype);
     List<RecordType> consumed = TestUtils.tryReadEntityOrLog(response, responseEntityType);
-    assertEquals(records.size(), consumed.size());
+    try {
+      assertEquals(records.size(), consumed.size());
+    } catch (AssertionError e) {
+      pause();
+      Response response2 = request(instanceUri + "/records").accept(accept).get();
+      assertOKResponse(response2, responseMediatype);
+      List<RecordType> consumed2 = TestUtils.tryReadEntityOrLog(response2, responseEntityType);
+      consumed.addAll(consumed2);
```

Comment:
nit: You can simplify `consumeMessages` by extracting this into a recursive method with an explicit recursion depth / counter argument.